[AArch64][GlobalISel] Prefer DUPLANE to REV

davemgreen opened the pull request:
Some shuffles containing undefs can match multiple instructions, such as <3,u,u,u> being either a duplane or a rev. This changes the order that different shuffles are considered, so that duplane is preferred which is simpler and more likely to lead to further combines.

c-rhodes reviewed (approved):
seems to have made thing slightly worse in `@shufflevector_v3p0` where there's now an extra inst (DUP), but overall seems like an improvement. I dont think the title is accurate though? It's not just REV it prefers DUPLANE over now

davemgreen committed:
[AArch64][GlobalISel] Prefer DUPLANE to REV

Some shuffles containing undefs can match multiple instructions, such as
<3,u,u,u> being either a duplane or a rev. This changes the order that
different shuffles are considered, so that duplane is preferred which is
simpler and more likely to lead to further combines.
